filebeat/input/filestream/fswatch.go

```diff
@@ -326,93 +358,134 @@ func (s *fileScanner) GetFiles() map[string]os.FileInfo {
 			continue
 		}
 
-		for _, file := range matches {
-			if s.shouldSkipFile(file) {
+		for _, filename := range matches {
+			it, err := s.getIngestTarget(filename)
+			if err != nil {
+				s.log.Debugf("cannot create an ingest target for file %q: %s", filename, err)
 				continue
 			}
 
-			// If symlink is enabled, it is checked that original is not part of same input
-			// If original is harvested by other input, states will potentially overwrite each other
-			if s.isOriginalAndSymlinkConfigured(file, uniqFileID) {
+			fd, err := s.toFileDescriptor(&it)
+			if err != nil {
+				s.log.Debugf("cannot create a file descriptor for an ingest target %q: %s", filename, err)
 				continue
 			}
 
-			fileInfo, err := os.Stat(file)
-			if err != nil {
-				s.log.Debug("stat(%s) failed: %s", file, err)
+			fileID := fd.FileID()
+			if knownFilename, exists := uniqueIDs[fileID]; exists {
+				s.log.Infof("%q points to an already known ingest target %q [%s==%s]. Skipping", fd.Filename, knownFilename, fileID, fileID)
```

Comment:
shall we promote it to `WARN` level? Can be potentially noisy but very useful for troubleshooting.

Reply:
I think `INFO` is fine. There's neither any degraded behavior nor is there necessarily any action a user could take.  And I know some customers are spooked by `WARN` level messages in the logs.  So I'd leave this as `INFO`.  We (maintainers) should be able to search for this message pretty easily during troubleshooting.